lib/action_subscriber/message_retry.rb

```diff
@@ -0,0 +1,38 @@
+module ActionSubscriber
+  module MessageRetry
+    def self.redeliver_message_with_backoff(env)
+      ttl = calculate_ttl(env)
+      return unless ttl
+      with_exchange(env, ttl) do |exchange|
+        exchange.publish(env.encoded_payload,
+          :content_type => env.content_type,
+          :routing_key => env.routing_key,
+          :expiration => ttl
+        )
+      end
+    end
+
+    # Private Implementation
+    def self.calculate_ttl(env)
+      death = env.headers.fetch("x-death", [{"original-expiration" => "20"}]).first
+      next_ttl = death["original-expiration"].to_s.to_i * 5
+      return false if next_ttl > 86_400_000
+      next_ttl
+    end
+
+    def self.with_exchange(env, ttl)
+      retry_name = "#{env.exchange}_retry_#{ttl}"
```

Comment:
Could a single retry exchange be used if each original queue has a retry queue in the retry exchange for every ttl? It might be easier introspect retries (through the web UI for example) if they're all happening under a single exchange, but maybe it's sixes?